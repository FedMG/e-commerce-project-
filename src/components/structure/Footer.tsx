import Link from "next/link";

const links = [
  {
    id: 1,
    title: "Policy",
    column: [
      { route: "Privacy Policy", path: "/privacy-policy" },
      { route: "Terms of Service", path: "/terms-of-service" },
      { route: "Consumer Rights", path: "/consumer-rights" },
    ],
  },
  {
    id: 2,
    title: "Support",
    column: [
      { route: "Contact Us", path: "/contact-us" },
      { route: "Forum", path: "/forum" },
      { route: "Common questions", path: "/common-questions" },
    ],
  },
  {
    id: 3,
    title: "Social Networks",
    column: [
      { route: "Instagram", path: "/instagram/ecommerce" },
      { route: "Twitter", path: "/twitter/ecommerce" },
      { route: "Facebook", path: "/facebook/ecommerce" },
    ],
  },
];

interface AnchorProps {
  path: string;
  route: string;
}

const Anchor: React.FC<AnchorProps> = ({ path, route }) => (
  <a
    href={path}
    className="text-white hover:text-gray-400"
    target="_blank"
    rel="noreferrer"
  >
    {route}
  </a>
);

export const Footer = () => {
  return (
    <footer className="fixed bottom-0 w-full bg-gray-900 text-white py-6">
      <div className="container mx-auto px-4">
        <div className="flex items-center justify-between">
          <div className="text-sm">
            <p>&copy; 2023 e-commerce</p>
          </div>
          <div className="flex flex-row sm:space-x-6">
            {links.map(({ id, title, column }) => (
              <div key={id}>
                <h4>{title}</h4>
                <div className="flex flex-col">
                  {column.map(({ route, path }) => {
                    if (title !== "Social Networks") {
                      return (
                        <Link
                          key={route}
                          href={path}
                          className="text-white hover:text-gray-400"
                        >
                          {route}
                        </Link>
                      );
                    }
                    return <Anchor key={route} path={path} route={route} />;
                  })}
                </div>
              </div>
            ))}
          </div>
        </div>
      </div>
    </footer>
  );
};
