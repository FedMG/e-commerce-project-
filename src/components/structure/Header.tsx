import { ReactElement } from 'react'
import Image from 'next/image'

export const Header = (): ReactElement => {
  return (
    <header className='flex bg-orange-300 border-gray-200'>
      <div className='flex items-center px-2 py-2'>
        <Image src='/logo.svg' alt='E-commerce logo' className='mr-3' width={35} height={35} />
        <span className='text-xl font-semibold'>
          E-commerce
        </span>
      </div>
    </header>
  )
}
