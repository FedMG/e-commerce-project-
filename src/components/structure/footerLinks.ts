import { FooterLinksColumns } from 'additional'

export const links: FooterLinksColumns[] = [
  {
    id: 1,
    title: 'Policy',
    column: [
      { route: 'Privacy Policy', path: '/privacy-policy' },
      { route: 'Terms of Service', path: '/terms-of-service' },
      { route: 'Consumer Rights', path: '/consumer-rights' }
    ]
  },
  {
    id: 2,
    title: 'Support',
    column: [
      { route: 'Forum', path: '/forum' },
      { route: 'Contact Us', path: '/contact-us' },
      { route: 'Common questions', path: '/common-questions' }
    ]
  },
  {
    id: 3,
    title: 'Networks',
    column: [
      { route: 'Twitter', path: '/twitter/ecommerce' },
      { route: 'Facebook', path: '/facebook/ecommerce' },
      { route: 'Instagram', path: '/instagram/ecommerce' }
    ]
  }
]
