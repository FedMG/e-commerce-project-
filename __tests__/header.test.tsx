import { render } from '@testing-library/react'

import { Header } from 'src/components/structure/Header'

it('should renders an header component unchanged', () => {
  const { container } = render(<Header />)
  expect(container).toMatchSnapshot()
})
