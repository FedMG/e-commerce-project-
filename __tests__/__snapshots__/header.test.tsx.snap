// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders an header component unchanged 1`] = `
<div>
  <header
    class="flex bg-orange-300 border-gray-200"
  >
    <div
      class="flex items-center px-2 py-2"
    >
      <img
        alt="E-commerce logo"
        class="mr-3"
        data-nimg="1"
        decoding="async"
        height="35"
        loading="lazy"
        src="/logo.svg"
        style="color: transparent;"
        width="35"
      />
      <span
        class="text-xl font-semibold"
      >
        E-commerce
      </span>
    </div>
  </header>
</div>
`;
